features/steps/steps.py

```diff
@@ -257,7 +268,7 @@ def step_impl(context, related, relating, other_entity, condition):
             for inst in context.model.by_type(related):
                 for rel in getattr(inst, 'Decomposes', []):
                     if not rel.RelatingObject.is_a(relating):
-                        errors.append(instance_structure_error(inst, rel.RelatingObject))
+                        errors.append(instance_structure_error(inst, [rel.RelatingObject], 'assigned to',{}))
```

Comment:
You already have `default_factory=dict`. I would leave it out here. Looks prettier.

Reply:
I have tried that, but ran into a 'class expected x arguments but got x-1 arguments' type of error. I don't really understand why and am looking into the dataclasses documentation/stackoverflow, but so far it is not clearer (maybe by modifying the __init__ function ourselves?)

Reply:
Fixed https://github.com/buildingSMART/ifc-gherkin-rules/commit/5267f10a101c67fa8d4ecdf01b761d9ce4ba48c0
Had to change 
`optional_values: field(default_factory=dict)`
to
`optional_values: dict = field(default_factory=dict)`

Reply:
Oh, I missed that, correct.